Unable to build with OTP 23

On Ubuntu 20.04 trying to use OTP v23 with jchash i get the following:

```
$ mix deps.compile jchash
===> Compiling jchash
make: Entering directory '/project/deps/jchash/c_src'
cc -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes -fPIC -I /usr/lib/erlang/erts-11.0.2/include/ -I /usr/lib/erlang/lib/erl_interface-4.0/include  -c -o /project/deps/jchash/c_src/jchash.o /project/deps/jchash/c_src/jchash.c
cc /project/deps/jchash/c_src/jchash.o -shared -L /usr/lib/erlang/lib/erl_interface-4.0/lib -lerl_interface -lei -o /project/deps/jchash/c_src/../priv/jchash.so
/usr/bin/ld: cannot find -lerl_interface
collect2: error: ld returned 1 exit status
make: *** [Makefile:59: /project/deps/jchash/c_src/../priv/jchash.so] Error 1
make: Leaving directory '/project/deps/jchash/c_src'
===> Hook for compile failed!

$ dpkg -l|grep esl-erlang
ii  esl-erlang                                 1:23.0.2-1

```

Is the deprecation of erl_interface the reason, mentioned [here](http://erlang.org/doc/general_info/deprecations.html#legacy-parts-of-erl_interface)?

--- a/c_src/Makefile
+++ b/c_src/Makefile
@@ -9,6 +9,7 @@ PROJECT := $(strip $(PROJECT))
 ERTS_INCLUDE_DIR ?= $(shell erl -noshell -s init stop -eval "io:format(\"~s/erts-~s/include/\", [code:root_dir(), erlang:system_info(version)]).")
 ERL_INTERFACE_INCLUDE_DIR ?= $(shell erl -noshell -s init stop -eval "io:format(\"~s\", [code:lib_dir(erl_interface, include)]).")
 ERL_INTERFACE_LIB_DIR ?= $(shell erl -noshell -s init stop -eval "io:format(\"~s\", [code:lib_dir(erl_interface, lib)]).")
+ERL_OTP ?= $(shell erl -noshell -s init stop -eval "io:format(\"~s\", [erlang:system_info(otp_release)]).")
 
 C_SRC_DIR = $(CURDIR)
 C_SRC_OUTPUT ?= $(CURDIR)/../priv/jchash.so
@@ -34,7 +35,10 @@ endif
 CFLAGS += -fPIC -I $(ERTS_INCLUDE_DIR) -I $(ERL_INTERFACE_INCLUDE_DIR)
 CXXFLAGS += -fPIC -I $(ERTS_INCLUDE_DIR) -I $(ERL_INTERFACE_INCLUDE_DIR)
 
-LDLIBS += -L $(ERL_INTERFACE_LIB_DIR) -lerl_interface -lei
+LDLIBS += -L $(ERL_INTERFACE_LIB_DIR) -lei
+ifeq ($(shell test $(ERL_OTP) -lt 23; echo $$?),0)
+	LDLIBS += -lerl_interface
+endif
 LDFLAGS += -shared
 
 # Verbosity.

